wcomponents-core/src/main/java/com/github/bordertech/wcomponents/AbstractWComponent.java

```diff
@@ -156,9 +156,10 @@ public void setIdName(final String idName) {
 					+ idName
 					+ " must start with a letter and followed by letters, digits and or underscores.");
 		}
-
-		ComponentModel model = getOrCreateComponentModel();
-		model.setIdName(idName);
+		if (getIdName() == null || !getIdName().equals(idName)) {
```

Comment:
Do we still want to create a new component mode if `getIdName() == null` *and* the idName arg is also null?

Reply:
In this particular instance an empty or null `idName` throws an IllegalArgumentException so it will never be null.